### DISCUSSION TITLE
Cross referencing from DSL to Other language(eg, .C,Cpp,.py)

### DISCUSSION THREAD

**Discussion by open-domain** [eclipse-langium/langium#890] [Q&A] [answered]
Hello Team

I have an usecase in my DSL, where I have to map/reference a variable/function from languages like Python,C,CPP(.py,.c,.cc,.cpp,.h,.hpp). Is it possible ? what would be your recommendation ?

MfG
Maha

**Comment by spoenemann** [comment, marked as answer]
That likely requires a lot of effort, as you'd need a parser for Python/C++ and integrate it into your DSL language server.

My recommendation: declare the variable/function in your DSL and generate Python/C/C++ files declaring the variable/function in those languages (e.g. C/C++ header file). Then you can reference the symbols both in the DSL and in the target languages without the need to parse each other's code.

**Reply by open-domain** [reply]
Thank you for your reply,
